training/src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -25,11 +28,31 @@
     @Autowired
     private BookRepository bookRepository;
 
+    @Autowired
+    private OpenLibraryService openLibraryService;
+
     @GetMapping
     public Iterable findAll() {
         return bookRepository.findAll();
     }
 
+    @GetMapping("/isbn/{isbn}")
+    public Book bookInfoByIsbn(@PathVariable String isbn)
+        throws BookNotFoundException, ParseException, JsonProcessingException {
+
+        Book bookInRepository = bookRepository.findFirstByIsbn(isbn);
+        if (bookInRepository == null) {
+            Book bookService = openLibraryService.bookInfo(("ISBN:").concat(isbn));
```

Comment:
The isbn concatenation should be done by the `openLibraryService.bookInfo` itself. It's something that you will always need to do and the controller should just pass the isbn and be abstracted from all the service logic.